hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -215,6 +222,28 @@ private OMResponse internalProcessRequest(OMRequest request) throws
         //  Added the assertion.
         assert (omClientRequest != null);
         OMClientRequest finalOmClientRequest = omClientRequest;
+
+        ClientInvocationId clientInvocationId = ClientInvocationId.valueOf(
+            ClientId.valueOf(UUID.nameUUIDFromBytes(ProtobufRpcEngine.Server.getClientId())),
+            ProtobufRpcEngine.Server.getCallId());
+        RetryCache.Entry cacheEntry =
+            omRatisServer.getServerDivision().getRetryCache().getIfPresent(clientInvocationId);
```

Comment:
The we can move the whole thing to `OzoneManagerRatisServer`.  It should check retry before creating `omClientRequest`.
```java
//OzoneManagerRatisServer
  public OMResponse checkRetryCache() throws ServiceException {
    final ClientInvocationId invocationId = ClientInvocationId.valueOf(getClientId(), getCallId());
    final RetryCache.Entry cacheEntry = getServerDivision().getRetryCache().getIfPresent(invocationId);
    if (cacheEntry == null) {
      return null;  //cache miss
    }
    //cache hit
    try {
      return getOMResponse(cacheEntry.getReplyFuture().get());
    } catch (ExecutionException ex) {
      throw new ServiceException(ex.getMessage(), ex);
    } catch (InterruptedException ex) {
      Thread.currentThread().interrupt();
      throw new ServiceException(ex.getMessage(), ex);
    }
  }
```